packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -509,7 +509,10 @@ class _TextSelectionHandleOverlayState extends State<_TextSelectionHandleOverlay
               child: widget.selectionControls.buildHandle(
                 context,
                 type,
-                widget.renderObject.size.height / widget.renderObject.maxLines,
+                widget.renderObject.maxLines != null
+                    ? widget.renderObject.size.height /
+                        widget.renderObject.maxLines
+                    : widget.renderObject.size.height,
```

Comment:
this should actually be the height of a line of text, which doesn't really have anything to do with the height of the render object...

Reply:
This is all kinds of wrong really. It's assuming that the text will be the same height on every line.

Reply:
As a first step, we should probably expose a getter on RenderEditable that returns the base height of each line (something like `text.style.fontSize * text.style.height * textScaleFactor`), then we should just use that getter here.